### PR TITLE
Sanitize meta values before passing to *_post_meta functions

### PR DESCRIPTION
**Pull request by jamesgol**
WordPress expects slashed data

## Description
Fixes issue #5366 

## How Has This Been Tested?
I can't be sure if this is going to break anything else, hopefully the tests cover enough.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.
- [ ] My code follows has proper inline documentation.

### CODE REVIEW

File: classes/PodsAPI.php [hunk 1]
```diff
@@ -310,6 +310,8 @@ public function save_post_meta( $id, $post_meta = null, $strict = false, $fields
 		}
 
 		foreach ( $post_meta as $meta_key => $meta_value ) {
+			// WordPress's *_post_meta() functions expect to receive 'slashed' data
+			$meta_value = pods_sanitize( $meta_value );
 			if ( null === $meta_value || ( $strict && '' === $post_meta[ $meta_key ] ) ) {
```
Comment (sc0ttkclark) [inline classes/PodsAPI.php]:
```suggestion
			if ( null === $meta_value || ( $strict && '' === $post_meta[ $meta_key ] ) ) {
```

File: classes/PodsAPI.php [hunk 2]
```diff
@@ -310,6 +310,8 @@ public function save_post_meta( $id, $post_meta = null, $strict = false, $fields
 		}
 
 		foreach ( $post_meta as $meta_key => $meta_value ) {
+			// WordPress's *_post_meta() functions expect to receive 'slashed' data
+			$meta_value = pods_sanitize( $meta_value );
```
Comment (sc0ttkclark) [inline classes/PodsAPI.php]:
```suggestion
			$meta_value = pods_sanitize( $meta_value );

```

File: classes/PodsAPI.php [hunk 3]
```diff
@@ -310,6 +310,9 @@ public function save_post_meta( $id, $post_meta = null, $strict = false, $fields
 		}
 
 		foreach ( $post_meta as $meta_key => $meta_value ) {
+			// WordPress's *_post_meta() functions expect to receive 'slashed' data
```
Comment (sc0ttkclark) [inline classes/PodsAPI.php]:
```suggestion
			// The WordPress *_post_meta() functions expect to receive 'slashed' data.
```

**Comment by sc0ttkclark** [comment]
I merged this too quickly, there's something that needs to be dug into further on this.

**Comment by sc0ttkclark** [comment]
I had to revert this merge

**Comment by jamesgol** [comment]
Oh no, what did it break?  I couldn't come up with anything in my testing

**Comment by sc0ttkclark** [comment]
I realized that PodsAPI::save_post() is unsanitizing the data, what we probably need to do here is to look at removing line 227 perhaps. Does that make sense on your side?

**Pull request merged**